src/providers/whatsapp/whatsapp.service.ts

```diff
@@ -17,59 +19,66 @@ export class WhatsappService {
   constructor(private readonly socketGateway: SocketGateway) {
     const appConfig = getAsyncAppConfig();
 
-    // create({
-    //   session: appConfig.appname,
-    //   logQR: false,
-    //   catchQR: this.onWaitQrCode,
-    // })
-    //   .then((client) => {
-    //     this.client = client;
-    //     this.socketGateway.broadcast('init', 'Sessão Criada');
-    //     console.log('client');
-    //   })
-    //   .catch(() => {
-    //     console.log('Erro ao criar instancia do whatsapp');
-    //   });
+    create({
+      session: appConfig.appname,
+      logQR: true,
+      catchQR: this.onWaitQrCode,
+    })
+      .then((client) => {
+        this.client = client;
+
+        process.on('SIGINT', function () {
+          client.close();
+        });
+      })
+      .catch(() => {
+        console.log('Erro ao criar instancia do whatsapp');
+      });
   }
 
   onWaitQrCode: CatchQR = (qrCode) => {
-    this.socketGateway.broadcast(
-      EventTypes.QRCODE,
-      'Aguardando leitura do QRCode',
-    );
-    console.log('qrCode');
+    this.socketGateway.broadcast(EventTypes.QR_CODE, qrCode);
   };
 
+  async listenOnConnection() {
+    if (await this.client.isConnected()) {
+      this.onConnection();
+    }
```

Comment:
```suggestion
    const isConnected = await this.client.isConnected()
    if (isConnected) {
      this.onConnection();
    }
```

é uma má pratica colocar o await direto dentro do if